lib/mongodb.js

```diff
@@ -1870,30 +1873,71 @@ MongoDB.prototype.ping = function(cb) {
   }
 };
 
+// Determine if a property must have an ObjectID value
+function mustBeObjectID(propDef) {
+  if (!propDef) return false;
+  if (propDef.mongodb && ObjectIdTypeRegex.test(propDef.mongodb.dataType)) return true;
+  return false;
+}
+
+// Determine if strictObjectIDCoercion should be enabled
+function enableStrictObjectIDCoercion(modelCtor, options) {
+  const settings = modelCtor.settings;
+  return (settings && settings.strictObjectIDCoercion) ||
+    (modelCtor.model && modelCtor.model.getConnector().settings.strictObjectIDCoercion) ||
```

Comment:
I find this unnecessarily complicated. Can you change `enableStrictObjectIDCoercion` into a connector method, so that you can access connector settings via `this.settings`? Obviously, other functions calling this one will have have to be converted into connector methods too.

Reply:
I agree, it is not simple, but it is not too complicated at the same time.

The following helper functions have to be added to the prototype to make `this.settings` possible:

1. `coercePropertyValue`
2. `coerceToObjectId`
3. `isObjectIDProperty`
4. `isStrictObjectIDCoercionEnabled`

Does the ability to use `this.settings` instead of `modelCtor.model && modelCtor.model.getConnector().settings`, out weight the addition of those extra methods to the prototype?

I initially started with the approach you suggested but decided otherwise, because I found it very inelegant in this case.

Reply:
Fair enough, I can live with the current proposal.